### PR TITLE
feat: Enable disabling Tensorboard logging [MLG-22]

### PR DESCRIPTION
**Pull request by szewaiyuen6**
### Unit tests
Added new unit tests

### End 2 end testing - Deepspeed

**Case 1: When logging is disabled**
Added line `self.context.set_enable_tensorboard_logging(False)` to the init function of the trial class in this [example](https://github.com/determined-ai/determined/tree/main/examples/deepspeed/cifar10_cpu_offloading).

Verified that:
(1) From the UI, tensorboard would not finish loading as there is no tensorboard data

(2) Also verified that the underlying s3 storage has no tensorboard file for this trial (experiment id is 906 and we see that no folder exist for it)
Screen Shot 2023-07-31 at 11 10 41 AM

Case 2: When logging is enabled (default behavior)
Using the [example](https://github.com/determined-ai/determined/tree/main/examples/deepspeed/cifar10_cpu_offloading) as is:
Verified that:
(1) From the UI, we can open tensorboard with metrics
Screen Shot 2023-07-31 at 11 07 21 AM

(2) Also verified that the underlying s3 storage has the tensorboard files for this trial (experiment id is 905)
Screen Shot 2023-07-31 at 11 10 41 AM

### End 2 end testing - Trainer API

**Case 1: logging disabled**
modified trainer [API example in the documentation](https://docs.determined.ai/latest/model-dev-guide/apis-howto/api-pytorch-ug.html#pytorch-trainer) adding this line and using a CIFAR10 PyTorch trial from other examples.
        
```
trial_inst = model.CIFARTrial(train_context)
trainer = det.pytorch.Trainer(trial_inst, train_context)
trainer.fit(
            ....
            enable_tensorboard_logging=False,
        )
```
Verified that:
(1) From the UI, tensorboard would not finish loading as there is no tensorboard data

(2) Also verified that the underlying s3 storage has no tensorboard file for this trial (experiment id is 904 and we see that no folder exist for it)
Screen Shot 2023-07-31 at 11 10 41 AM

** Case 2: logging enabled**
Same setup as above except that we do not include the line `enable_tensorboard_logging=False,` when calling trainer.fit.
Default behavior should enable logging to tensorboard.

verified that:
(1) from the UI, we can open tensorboard
Screen Shot 2023-07-31 at 11 41 23 AM

(2) We see corresponding folder in underlying TB s3 storage (the experiment id is 907)
Screen Shot 2023-07-31 at 11 42 03 AM

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: harness/determined/pytorch/_trainer.py [hunk 1]
```diff
@@ -72,6 +72,7 @@ def fit(
         latest_checkpoint: Optional[str] = None,
         step_zero_validation: bool = False,
         test_mode: bool = False,
+        enable_tensorboard_logging: bool = True,
```
Comment (azhou-determined) [inline harness/determined/pytorch/_trainer.py:75]: if this is on the context, it might be a better fit for `_trainer.init`?

Comment (szewaiyuen6) [reply harness/determined/pytorch/_trainer.py:75]: I don't have a strong opinion here, so happy to go either way.

Following are my original reason to put it under fit:
If we look at the arguments under init vs fit, I just think this new option seem more similar with arguments of fit vs init (e.g. step_zero_validation, test_mode) whereas arguments on init are more like exp_config, hparams, etc.

Comment (azhou-determined) [reply harness/determined/pytorch/_trainer.py:75]: i'm leaning towards putting it in the context's `init` because it seems like this value is specific to the "runtime context" of the training loop, and doesn't really have to do with the actual `.fit()` training loop configurations. feels like it belongs more with `debug_enabled`, `managed_training` in the context (though currently not configurable/passed in by the user), rather than training loop arguments like reporting/validation steps. 

i feel it's also easier to tell where/when the context values are being set if we try and keep them in the same place.

File: harness/determined/pytorch/_trainer.py [hunk 2]
```diff
@@ -289,4 +291,10 @@ def init(
             debug_enabled=debug_enabled,
         )
 
+        # Tensorboard logging is enabled by default
+        # We only need to call set_enable_tensorboard_logging on the trial object of user pass in
+        # enable_tensorboard_logging = False
+        if not enable_tensorboard_logging:
```
Comment (azhou-determined) [inline harness/determined/pytorch/_trainer.py:297]: pass this into the `PyTorchTrialContext` constructor directly? part of the reason to move to `init` is to keep the context initialization in the same place.

**Review by azhou-determined — APPROVED** [review]
lgtm
